lua/lsp/init.lua

```diff
@@ -65,11 +65,34 @@ end
 -- Set Default Prefix.
 -- Note: You can set a prefix per lsp server in the lv-globals.lua file
 function lsp_config.setup_handlers()
-  vim.lsp.handlers["textDocument/publishDiagnostics"] = vim.lsp.with(vim.lsp.diagnostic.on_publish_diagnostics, {
-    virtual_text = lvim.lsp.diagnostics.virtual_text,
-    signs = lvim.lsp.diagnostics.signs,
-    underline = lvim.lsp.document_highlight,
-  })
+  vim.lsp.handlers["textDocument/publishDiagnostics"] = function(_, _, params, client_id, _)
+    local config = { -- your config
+      virtual_text = lvim.lsp.diagnostics.virtual_text,
+      signs = lvim.lsp.diagnostics.signs,
+      underline = lvim.lsp.diagnostics.underline,
+      update_in_insert = lvim.lsp.diagnostics.update_in_insert,
+    }
+    local uri = params.uri
+    local bufnr = vim.uri_to_bufnr(uri)
+
+    if not bufnr then
+      return
+    end
+
+    local diagnostics = params.diagnostics
+
+    for i, v in ipairs(diagnostics) do
+      diagnostics[i].message = string.format("%s: %s", v.source, v.message)
```

Comment:
Have you considered adding the code too ? Maybe smth like `"%s [%s]: %s"`
See [here](https://github.com/jose-elias-alvarez/null-ls.nvim/pull/57/files#diff-c2c076648e1b1e04ff69ac1030444a8ffbfda480d12fe8771b719a77cba92d7fR142) for examples of code values.

Reply:
> Have you considered adding the code too ? Maybe smth like `"%s [%s]: %s"`
> See [here](https://github.com/jose-elias-alvarez/null-ls.nvim/pull/57/files#diff-c2c076648e1b1e04ff69ac1030444a8ffbfda480d12fe8771b719a77cba92d7fR142) for examples of code values.

I'll try to see what I can come up with.
Thanks for all the amazing work for null-ls btw :)